multicore_runtime/wsgi_config.py

```diff
@@ -0,0 +1,216 @@
+# Copyright 2015 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Utilities to configure the dispatcher, middleware and environment."""
+
+import logging
+import os
+import threading
+import UserDict
+
+from static_files import static_app_for_regex_and_files
+
+from google.appengine.api import appinfo
+from google.appengine.api import appinfo_includes
+from google.appengine.runtime import wsgi
+
+
+def get_module_config_filename():
+  """Returns the path of the module configuration file (e.g. 'app.yaml').
+
+  Returns:
+    The path of the module configuration file.
+
+  Raises:
+    KeyError: The MODULE_YAML_PATH environment variable is not set.
+  """
+  module_yaml_path = os.environ['MODULE_YAML_PATH']
+  logging.info('Using module_yaml_path from env: %s', module_yaml_path)
+  return module_yaml_path
+
+
+def get_module_config(filename):
+  """Returns the parsed module config."""
+  with open(filename) as f:
+    return appinfo_includes.Parse(f)
+
+
+def app_for_script(script):
+  """Returns the WSGI app specified in the input string, or None on failure."""
+  try:
+    app, unused_filename, err = wsgi.LoadObject(script)
+    if err:
+      raise err
```

Comment:
So what type of value is `err` here?

Is it really an Exception object?

I'd find it weird if it is... If `wsgi.LoadObject` runs into an error, why would it return an Exception object as part as a tuple instead of just `raise`ing it?